src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BedSearchService.kt

```diff
@@ -118,10 +123,33 @@ class BedSearchService(
 
         val results = candidateResults.filter { !bedsWithABookingInTurnaround.contains(it.bedId) }
 
+        val distinctIds = candidateResults.map { it.premisesId }.distinct()
+        val overlappedBookings = bookingRepository.findAllByPremisesIdsAndOverLappingDate(distinctIds, startDate, endDate)
+        val groupedOverlappedBookings = overlappedBookings
+          .map { transformBookingToOverlapResult(it, startDate, endDate) }
+          .groupBy { it.premisesId }
+
+        results.forEach {
+          val overlappingBookings = groupedOverlappedBookings[it.premisesId]?.toList() ?: listOf()
+          it.overlaps.addAll(overlappingBookings)
+        }
+
         return@validated success(
           results,
         )
       },
     )
   }
+
+  fun transformBookingToOverlapResult(booking: BookingEntity, startDate: LocalDate, endDate: LocalDate): TemporaryAccommodationBedSearchOverlapResult {
+    val queryDuration = startDate..endDate
+    val bookingDuration = booking.arrivalDate..booking.departureDate
+
+    return TemporaryAccommodationBedSearchOverlapResult(
+      crn = booking.crn,
+      days = bookingDuration countOverlappingDays queryDuration,
```

Comment:
Is this one of those infix functions again? The lack of parenthesis always trips me up.

Reply:
Saw that it is later on!